### PR TITLE
Resize large color image to improve rendering time

### PR DESCRIPTION
**Pull request by hhsttkh**
Please review this request.

### CODE REVIEW

**Comment by yurydelendik** [comment]
It will make #7041 worse

**Comment by Hengjie** [comment]
@yurydelendik that's weird because this PR's code path is not used when it tries to render #7041 https://github.com/mozilla/pdf.js/files/150290/default.0002.pdf

**Comment by fkaelberer** [comment]
@Hengjie: I don't think @yurydelendik has tried this, but rather refers to the weak scaling algorithm that is used (see discussion in #7047).

File: src/core/image.js [hunk 1]
```diff
@@ -534,6 +534,80 @@ var PDFImage = (function PDFImageClosure() {
...
+        var i, j, py, newIndex = 0,
+          oldIndex;
+        var w1Scanline = w1 * components;
+        if (alpha01 !== 1) {
+          alpha01 = 0;
+        }
+
+        if (components === 1) {
+          for (i = 0; i < h2; i++) {
+            py = Math.floor(i * yRatio) * w1Scanline;
```
Comment (fkaelberer) [inline src/core/image.js]: `yRatio` is not initialized.

**Comment by timvandermeij** [comment]
Closing since this is currently not in a mergeable state, it may cause image quality issues and it's not clear if it's even necessary if #9255 is finished. Let's await that and if this is still a problem we can always reopen/revisit. Thanks.

**Pull request closed without merging**